skore/tests/unit/reports/estimator/feature_importance/test_permutation_importance.py

```diff
@@ -339,3 +341,125 @@ def test_not_fitted(regression_data):
+
+    def test_non_pipeline(self, data):
+        """
+        For non-pipeline estimators, changing at_step should not change the results.
+        """
+        X, y = data
+        report = EstimatorReport(
+            LinearRegression(), X_train=X, y_train=y, X_test=X, y_test=y
+        )
+
+        result_start = report.feature_importance.permutation(seed=42, at_step=0)
+        result_end = report.feature_importance.permutation(seed=42, at_step=-1)
+
+        pd.testing.assert_frame_equal(result_start, result_end)
+
+    AT_STEP_TOO_LARGE = (
+        "at_step must be strictly smaller in magnitude than "
+        "the number of steps in the Pipeline"
+    )
+
+    @pytest.mark.parametrize(
+        "at_step, err_msg",
+        [
+            (8, AT_STEP_TOO_LARGE),
+            (-8, AT_STEP_TOO_LARGE),
+            ("hello", "'hello' is not in list"),
+            (0.5, "at_step must be an integer or a string"),
+        ],
+    )
+    def test_value_error(self, pipeline_report, at_step, err_msg):
+        """If `at_step` value is not appropriate, a ValueError is raised."""
+        with pytest.raises(ValueError, match=err_msg):
+            pipeline_report.feature_importance.permutation(seed=42, at_step=at_step)
+
+    def test_sparse_array(self, data):
+        """If one of the steps outputs a sparse array, `permutation` still works."""
+        X, y = data
+        pipeline = make_pipeline(
+            SplineTransformer(sparse_output=True), LinearRegression()
+        )
+        report = EstimatorReport(pipeline, X_train=X, X_test=X, y_train=y, y_test=y)
+
+        report.feature_importance.permutation(seed=42, at_step=-1)
+
+    def test_feature_names(self, data):
+        """If the requested pipeline step gives proper feature names,
+        these names should appear in the output."""
+        X, y = data
+        pipeline = make_pipeline(SplineTransformer(), LinearRegression())
+        report = EstimatorReport(pipeline, X_train=X, X_test=X, y_train=y, y_test=y)
+
+        result = report.feature_importance.permutation(seed=42, at_step=-1)
+        last_step_feature_names = list(report.estimator_[0].get_feature_names_out())
+        assert list(result.index.levels[1]) == last_step_feature_names
+
+    def test_non_sklearn_pipeline(self, data):
+        """If the pipeline contains non-sklearn-compliant transformers,
+        `permutation` still works."""
+
+        class Scaler(TransformerMixin, BaseEstimator):
+            def fit(self, X, y=None):
+                self.mean_ = X.mean(axis=0)
+                self.std_ = X.std(axis=0)
+                return self
+
+            def transform(self, X, y=None):
+                X_ = X.copy()
+                return (X_ - self.mean_) / self.std_
+
+        X, y = data
+        pipeline = make_pipeline(Scaler(), LinearRegression())
+        report = EstimatorReport(pipeline, X_train=X, y_train=y, X_test=X, y_test=y)
+
+        report.feature_importance.permutation(seed=42, at_step=-1)
```

Comment:
```suggestion
        report.feature_importance.permutation(seed=42, at_step=at_step)
```

Reply:
It would also be nice to check the feature names that we get instead to only have a smoke test here.